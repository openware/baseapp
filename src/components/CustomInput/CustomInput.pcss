.custom-input {
  input {
    border-color: var(--input-border-color);
  }

  label {
    background: var(--input-background-color);
    font-size: calc(var(--gap) * 2);
    left: 5%;
    margin-top: -8px;
    padding: 0 calc(var(--gap) * 0.7);
    position: absolute;
    top: 0;
    z-index: 5;
  }

  input,
  input:focus {
    background: var(--input-background-color) !important;
    box-shadow: none;
    color: var(--input-text-color);
  }

  input:focus-within {
    border-color: var(--input-border-color-active);

    label {
      color: var(--input-label-color-active);
    }
  }

  input:-webkit-autofill,
  input:-webkit-autofill:hover,
  input:-webkit-autofill:focus {
    border: 1px solid var(--input-border-color);
    -webkit-box-shadow: 0 0 0 1000px var(--input-background-color) inset;
    -webkit-text-fill-color: var(--input-text-color);
    transition: background-color 5000s ease-in-out 0s;
  }
}
