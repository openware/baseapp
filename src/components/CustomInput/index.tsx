import { Input } from '@openware/components';
import * as React from 'react';

export interface CustomInputProps {
    type: string;
    label: string;
    defaultLabel: string;
    handleChangeInput: (value: string) => void;
    inputValue: string;
    handleFocusInput: () => void;
    classNameLabel: string;
    classNameInput: string;
    placeholder: string;
    autoFocus?: boolean;
    onKeyPress?: (event: React.KeyboardEvent<HTMLInputElement>) => void;
}

class CustomInput extends React.Component<CustomInputProps> {
    public render() {
        const {
            label,
            placeholder,
            defaultLabel,
            inputValue,
            classNameLabel,
            classNameInput,
            type,
            autoFocus,
            onKeyPress,
        } = this.props;

        return (
            <React.Fragment>
                <label className={classNameLabel}>
                    {inputValue && (label || defaultLabel)}
                </label>
                <Input
                    type={type}
                    value={inputValue}
                    placeholder={placeholder}
                    className={classNameInput}
                    onFocus={this.props.handleFocusInput}
                    onBlur={this.props.handleFocusInput}
                    onChangeValue={this.props.handleChangeInput}
                    autoFocus={autoFocus}
                    onKeyPress={onKeyPress}
                />
            </React.Fragment>
        );
    }
}

export {
    CustomInput,
};
