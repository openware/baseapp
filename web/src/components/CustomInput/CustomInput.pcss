.custom-input {
  input {
    border-color: var(--input-border-color);
  }

  label {
    background: var(--input-background-color);
    font-size: calc(var(--gap) * 2);
    left: 5%;
    margin-top: -8px;
    padding: 0 calc(var(--gap) * 0.7);
    position: absolute;
    top: 0;
    z-index: 5;
  }

  input,
  input:focus {
    background: var(--input-background-color) !important;
    box-shadow: none;
    color: var(--input-text-color);
  }

  input:focus {
    background: var(--input-background-color-active) !important;
  }

  input:focus-within {
    border-color: var(--input-border-color-active);

    label {
      color: var(--input-label-color-active);
    }
  }

  input:-webkit-autofill,
  input:-webkit-autofill:hover,
  input:-webkit-autofill:focus {
    border: 2px solid var(--input-border-color);
    -webkit-box-shadow: 0 0 0 1000px var(--input-background-color) inset;
    -webkit-text-fill-color: var(--input-text-color);
    transition: background-color 5000s ease-in-out 0s;
  }
}

.pg-mobile-app .custom-input {
  font-size: 12px;

  input {
    font-size: 16px;

    &::placeholder {
      font-size: 14px;
    }
  }
}
