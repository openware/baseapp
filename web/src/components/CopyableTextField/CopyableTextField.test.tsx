import { shallow } from 'enzyme';
import * as React from 'react';
import { CustomInput } from '../CustomInput';
import { CopyableTextField, CopyableTextFieldProps } from './';


const defaultProps: CopyableTextFieldProps = {
    value: '1F1tAaz5x1HUXrCNLbtMDqcw6o5GNn4dfE',
    fieldId: 'copy_id',
};

const setup = (props: Partial<CopyableTextFieldProps> = {}) =>
    shallow(<CopyableTextField {...{ ...defaultProps, ...props }} />);

describe('CopyableTextField', () => {
    it('should render', () => {
        const wrapper = setup();
        expect(wrapper).toMatchSnapshot();
    });

    it('should contain cr-copyable-text-field className', () => {
        const wrapper = setup();
        expect(wrapper.find('.cr-copyable-text-field')).toHaveLength(1);
    });

    it('should render 1 input tag', () => {
        const wrapper = setup();
        const input = wrapper.find(CustomInput);
        expect(input.length).toBe(1);
    });

});
