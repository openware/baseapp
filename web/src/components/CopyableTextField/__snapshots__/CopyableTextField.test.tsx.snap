// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CopyableTextField should render 1`] = `
<div
  className="cr-copyable-text-field"
>
  <InputGroup>
    <CustomInput
      defaultLabel=""
      handleClick={[Function]}
      id="copy_id"
      inputValue="1F1tAaz5x1HUXrCNLbtMDqcw6o5GNn4dfE"
      label=""
      placeholder=""
      readOnly={true}
      type="text"
    />
    <InputGroupAppend>
      <Button
        active={false}
        disabled={false}
        onClick={[Function]}
        size="lg"
        variant="primary"
      >
        Copy
      </Button>
    </InputGroupAppend>
  </InputGroup>
</div>
`;
